ui/uihandler.go

```diff
@@ -155,98 +194,106 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 			StartTime                   string
 			TargetURL                   string
 			Labels                      string
+			RunID                       int64
 			UpTime                      time.Duration
 			TestExpectedDurationSeconds float64
 			Port                        int
-			DoExit                      bool
+			DoStop                      bool
 			DoLoad                      bool
 		}{r, opts.GetHeaders(), periodic.Version, logoPath, debugPath, chartJSPath,
-			startTime.Format(time.ANSIC), url, labels,
-			fhttp.RoundDuration(time.Since(startTime)), dur.Seconds(), httpPort, DoExit, DoLoad})
+			startTime.Format(time.ANSIC), url, labels, runid,
+			fhttp.RoundDuration(time.Since(startTime)), dur.Seconds(), httpPort, mode == stop, mode == run})
 		if err != nil {
 			log.Critf("Template execution failed: %v", err)
 		}
-
-		if !DoLoad && !DoExit {
-			return
+	}
+	switch mode {
+	case menu:
+		// nothing more to do
+	case stop:
+		if runid <= 0 { // Stop all
+			i := 0
+			uiRunMapMutex.Lock()
+			for _, v := range runs {
+				v.Abort()
+				i++
+			}
+			uiRunMapMutex.Unlock()
+			log.Infof("Interrupted %d runs", i)
+		} else { // Stop one
+			uiRunMapMutex.Lock()
+			v := runs[runid]
+			if v != nil {
```

Comment:
Change to
v, ok := runs[runid]
 			if ok {

Reply:
k